packages/console/src/pages/EnterpriseSso/Guide/XmlFileUploader/index.tsx

```diff
@@ -0,0 +1,92 @@
+import classNames from 'classnames';
+import { useState, useRef, type ChangeEvent, useEffect } from 'react';
+
+import Delete from '@/assets/icons/delete.svg';
+import FileIcon from '@/assets/icons/file-icon.svg';
+import UploaderIcon from '@/assets/icons/upload.svg';
+import Button from '@/ds-components/Button';
+
+import * as styles from './index.module.scss';
+
+type FileInfo = {
+  name: string;
+  size: number; // In bytes.
+};
+
+type Props = {
+  onChange: (file?: File) => void;
+  value?: File;
+};
+
+function XmlFileUploader({ onChange, value }: Props) {
+  const [fileInfo, setFileInfo] = useState<FileInfo>();
```

Comment:
This state might not be needed, since you always emit `file` object to the parent component on input change

Reply:
Good call, removed the state to make it simpler